File: src/com/l2kt/gameserver/skills/funcs/FuncAtkAccuracy.java
package com.l2kt.gameserver.skills.funcs;

import com.l2kt.gameserver.model.actor.Summon;
import com.l2kt.gameserver.skills.Env;
import com.l2kt.gameserver.skills.Formulas;
import com.l2kt.gameserver.skills.Stats;
import com.l2kt.gameserver.skills.basefuncs.Func;

public class FuncAtkAccuracy extends Func
{
	static final FuncAtkAccuracy _faa_instance = new FuncAtkAccuracy();
	
	public static Func getInstance()
	{
		return _faa_instance;
	}
	
	private FuncAtkAccuracy()
	{
		super(Stats.ACCURACY_COMBAT, 0x10, null, null);
	}
	
	@Override
	public void calc(Env env)
	{
		final int level = env.getCharacter().getLevel();
		
		env.addValue(Formulas.BASE_EVASION_ACCURACY[env.getCharacter().getDEX()] + level);
		
		if (env.getCharacter() instanceof Summon)
			env.addValue((level < 60) ? 4 : 5);
	}
}

File: src/com/l2kt/gameserver/skills/funcs/FuncAtkAccuracy.kt
package com.l2kt.gameserver.skills.funcs

import com.l2kt.gameserver.model.actor.Summon
import com.l2kt.gameserver.skills.Env
import com.l2kt.gameserver.skills.Formulas
import com.l2kt.gameserver.skills.Stats
import com.l2kt.gameserver.skills.basefuncs.Func

object FuncAtkAccuracy : Func(Stats.ACCURACY_COMBAT, 0x10, null, null) {

    override fun calc(env: Env) {
        val level = env.character!!.level

        env.addValue(Formulas.BASE_EVASION_ACCURACY[env.character!!.dex] + level)

        if (env.character is Summon)
            env.addValue((if (level < 60) 4 else 5).toDouble())
    }
}